docs/guides/how-to-write-a-gadget.md

```diff
@@ -0,0 +1,141 @@
+# How to write a Gadget?
```

Comment:
Please remove the title and add a header like this generate this on our website. 

```
---
title: How to write a Gadget?
weight: 100
description: >
  How to write a Gadget?
---
```